### PR TITLE
Feature/settings fe

### PR DESCRIPTION
**Pull request by Eliory09**
Issue #304 :
User settings page frontend.

- HTML, CSS, JS.
- Modified to work with the current design.

### CODE REVIEW

File: app/static/grid_style.css [hunk 1]
```diff
@@ -94,7 +94,7 @@ header {
   position: sticky;
   top: 0;
   display: flex;
-  grid-flow: row wrap;
+  /* grid-flow: row wrap; */
```
Comment (ivarshav) [inline app/static/grid_style.css]: remove?

File: app/static/settings_style.css [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+.settings-options ul li a {
+  color: #222831;
+  transition: all .3s ease-in-out;
+  cursor: pointer;
+  text-decoration: none;
+}
+
+.settings-options ul li a:hover {
+  color: #5786f5;
```
Comment (ivarshav) [inline app/static/settings_style.css]: You can combine with the active def below (`.settings-options ul li a.active`)

File: app/static/settings_style.css [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+#select-form-settings {
+  font-size: 0.8rem;
+}
+
+/* For Mobile */
+@media screen and (max-width: 700px) { 
+  .settings {
+    display: block;
+    font-size: 3.5vw;
```
Comment (ivarshav) [inline app/static/settings_style.css]: what is vw?

File: app/static/settings_style.css [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+  .settings-main .form-select {
+    width: 250px;
+  }
+
+  .settings-main h2 {
+    margin-bottom: 1rem;
+  }
+    
+  .settings-main p {
+    margin: 1rem 0px;
```
Comment (ivarshav) [inline app/static/settings_style.css]: prefer not to use px :)

File: app/templates/settings.html [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+      </header>
+        <main class="settings">
+          <div id="settings-left">
+            <!-- Headline -->
+            <!-- <div class="settings-headline">
+                <h1>Settings</h1>
+            </div> -->
+            <section class="settings-layout">
+              <div class="left-options-bar">
+                <!-- <div class="long-slider">
```
Comment (ivarshav) [inline app/templates/settings.html]: Do you want to keep this?

File: app/templates/settings.html [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+                <label class="form-check-label" for="view-tick3">
+                  View calendars side by side in day view
+                </label>
+              </div>
+            </article>
+          </div>
+        </main>
+      
+      
+      <script>
```
Comment (ivarshav) [inline app/templates/settings.html]: Consider add a js file for that

File: app/static/grid_style.css (hunk 1 above)
Comment (Eliory09) [reply app/static/grid_style.css]: Removed. Thanks.

File: app/static/settings_style.css (hunk 3 above)
Comment (Eliory09) [reply app/static/settings_style.css]: Viewport units. Tested it on mobile view and its seems to be fine.

File: app/static/settings_style.css (hunk 4 above)
Comment (Eliory09) [reply app/static/settings_style.css]: You're right :)

File: app/templates/settings.html (hunk 5 above)
Comment (Eliory09) [reply app/templates/settings.html]: No :)

File: app/templates/settings.html (hunk 6 above)
Comment (Eliory09) [reply app/templates/settings.html]: Separated into a .js file.

**Comment by Eliory09** [comment]
@yammesicka ,
I'd really appreciate if you'll review my user settings page frontend, so I could commit changes to the code as fast as possible.
Settings team rely on this page for further development.
Thanks in advance!

File: app/static/settings_style.css [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+.settings {
+  color: #222831;
+  display: flex;
+  font-size: 0.9rem;
+}
+
+#settings-left {
+  padding: 1.5rem 1.5rem;
```
Comment (yammesicka) [inline app/static/settings_style.css]: Prefer working with em wherever possible.

File: app/static/settings_style.css [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+  display: flex;
+  flex-direction: column;
+  background-color: rgb(230, 230, 230)
+}
+
+.settings-headline {
+  display: flex;
+  width: 200px;
```
Comment (yammesicka) [inline app/static/settings_style.css]: em?

File: app/static/settings_style.css [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+.settings-options ul {
+  list-style-type: none;
+  padding-left: 0;
+  padding-right: 1rem;
+}
+
+.settings-options ul li {
+  margin-bottom: 15px;
```
Comment (yammesicka) [inline app/static/settings_style.css]: em

File: app/static/settings_style.css [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+.settings-main h2 {
+  margin-bottom: 1rem;
+}
+
+.settings-main p {
+  margin: 1rem 0rem;
+}
+
+.settings-main .form-select {
+  width: 250px;
```
Comment (yammesicka) [inline app/static/settings_style.css]: em

File: app/static/settings_style.css [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+  .settings-options {
+    margin-bottom: 30px;
+  }
+
+  .settings-options ul {
+    padding-left: 0;
+  }
+
+  .settings-options ul li {
+    margin-bottom: 5px;
```
Comment (yammesicka) [inline app/static/settings_style.css]: em

File: app/static/settings_style.css [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+  .settings-main .form-select {
+    width: 250px;
+  }
+
+  .settings-main h2 {
+    margin-bottom: 1rem;
+  }
+    
+  .settings-main p {
+    margin: 1rem 0px;
```
Comment (yammesicka) [inline app/static/settings_style.css]: em

File: app/static/js/settings.js [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+  tab_elem = document.getElementById(tab_id);
+  tab_num = parseInt(tab_id.slice(-1));
+  tab[tab_num].style.display = "block";
+  tab_elem.classList.add("active");
+
+  var els = document.getElementById("options").children;
+  for (var i=0, l=els.length; i<l; i++) {
+  if (els[i].childNodes[0].id != tab_id) {
+    els[i].childNodes[0].classList.remove("active")
+  }}};
```
Comment (yammesicka) [inline app/static/js/settings.js]: Reformat using some JS prettifier.

File: app/static/js/settings.js [hunk 14]
```diff
@@ -0,0 +1,26 @@
+const tabBtn =
+  document.querySelectorAll(".tab");
+  const tab =
+  document.querySelectorAll(".tabShow");
+
+  function clearMenu(options) {
+  for (var i=0, l=options.length; i<l; i++) {
+    cat = document.getElementById("options").children[i].children[0]
+    if (!(cat.classList.contains("active"))) {
+    options[i].style.display = "none";
```
Comment (yammesicka) [inline app/static/js/settings.js]: Prefer removing/adding classes

**Review by yammesicka — CHANGES_REQUESTED** [review]
Great start, we should refine the CSS/JS

File: app/static/settings_style.css (hunk 7 above)
Comment (Eliory09) [reply app/static/settings_style.css]: @aviadamar told me to use rem, it's his fault!
סתם! תיקנתי :)

File: app/static/settings_style.css (hunk 12 above)
Comment (Eliory09) [reply app/static/settings_style.css]: I don't know if I got it right, but I think I should use em.

File: app/static/js/settings.js (hunk 13 above)
Comment (Eliory09) [reply app/static/js/settings.js]: Done!

File: app/static/js/settings.js (hunk 14 above)
Comment (Eliory09) [reply app/static/js/settings.js]: Fixed

File: app/routers/settings.py [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+
+router = APIRouter(
+    prefix="/settings",
+    tags=["settings"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.get("/")
+async def settings(request: Request):
```
Comment (Gonzom) [inline app/routers/settings.py]: return type

File: app/static/settings_style.css [hunk 16]
```diff
@@ -0,0 +1,149 @@
...
+  transition: all .3s ease-in-out;
+  cursor: pointer;
+  text-decoration: none;
+}
+
+.settings-options ul li a:hover, .settings-options ul li a.active {
+  color: #5786f5;
+}
+
+.tabShow {
```
Comment (Gonzom) [inline app/static/settings_style.css]: names are lowercase and separated by "-".

File: app/static/settings_style.css [hunk 17]
```diff
@@ -0,0 +1,149 @@
...
+  display: block;
+  transition: all .5s ease-in;
+}
+
+.tabHide {
```
Comment (Gonzom) [inline app/static/settings_style.css]: names are lowercase and separated by "-".

File: app/static/settings_style.css [hunk 18]
```diff
@@ -0,0 +1,149 @@
...
+
+.settings-main p {
+  margin: 1em 0em;
+}
+
+.settings-main .form-select {
+  width: 250px;
+}
+
+.tabShow p {
```
Comment (Gonzom) [inline app/static/settings_style.css]: names are lowercase and separated by "-".

File: app/static/settings_style.css [hunk 19]
```diff
@@ -0,0 +1,149 @@
...
+  display: block;
+  align-self: left;
+}
+
+.tabShow h2 {
```
Comment (Gonzom) [inline app/static/settings_style.css]: names are lowercase and separated by "-".

File: app/routers/settings.py (hunk 15 above)
Comment (Eliory09) [reply app/routers/settings.py]: Your'e right. Thanks!

File: app/static/settings_style.css (hunk 19 above)
Comment (Eliory09) [reply app/static/settings_style.css]: Thanks for the review! All typos fixed.

File: app/static/js/settings.js [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+    let selected_tab_content = document.querySelector("#" + tab_id + "-content");
+    shown_tab.classList.remove("tab-show");
+    shown_tab.classList.add("tab-hide");
+    for (btn of tabBtn) {
+        btn.children[0].classList.remove("active");
+    }
+    document.getElementById(tab_id).classList.add("active");
+    selected_tab_content.classList.remove("tab-hide");
+    selected_tab_content.classList.add("tab-show");
+};
```
Comment (yammesicka) [inline app/static/js/settings.js]: The ; here is unnecessary

File: app/static/js/settings.js [hunk 21]
```diff
@@ -0,0 +1,24 @@
+const tabBtn =
+    document.querySelectorAll(".tab");
```
Comment (yammesicka) [inline app/static/js/settings.js]: Keep it in the same line

File: app/static/js/settings.js [hunk 22]
```diff
@@ -0,0 +1,24 @@
...
+
+
+function tabClick(tab_id) {
+    let shown_tab = document.querySelector(".tab-show");
```
Comment (yammesicka) [inline app/static/js/settings.js]: shownTab

File: app/static/js/settings.js [hunk 23]
```diff
@@ -0,0 +1,24 @@
...
+    let selected_tab_content = document.querySelector("#" + tab_id + "-content");
```
Comment (yammesicka) [inline app/static/js/settings.js]: selectedTabContent

Comment (yammesicka) [reply app/static/js/settings.js]: getElementById

Comment (yammesicka) [reply app/static/js/settings.js]:
```
`${tabId}-content`
```

File: app/static/js/settings.js [hunk 24]
```diff
@@ -0,0 +1,24 @@
...
+        btn.children[0].classList.remove("active");
+    }
+    document.getElementById(tab_id).classList.add("active");
+    selected_tab_content.classList.remove("tab-hide");
+    selected_tab_content.classList.add("tab-show");
+};
+
+
+for (i = 0; i < tabBtn.length; i++) {
+    let btn = document.getElementById("tab" + i);
```
Comment (yammesicka) [inline app/static/js/settings.js]: `const`

File: app/static/js/settings.js [hunk 25]
```diff
@@ -0,0 +1,24 @@
+const tabBtn =
```
Comment (yammesicka) [inline app/static/js/settings.js]: This, and the for in the last lines of this file, should only run after you ensure the page is fully loaded (window -> load or document's DOMContentLoaded)

File: app/static/settings_style.css (hunk 8 above)
Comment (yammesicka) [reply app/static/settings_style.css]: em

File: app/static/settings_style.css (hunk 9 above)
Comment (yammesicka) [reply app/static/settings_style.css]: em

File: app/static/settings_style.css (hunk 2 above)
Comment (yammesicka) [reply app/static/settings_style.css]: yup

File: app/static/settings_style.css (hunk 10 above)
Comment (yammesicka) [reply app/static/settings_style.css]: em

File: app/templates/partials/calendar/navigation.html [hunk 26]
```diff
@@ -16,7 +15,9 @@
         <ion-icon name="search" alt="Search icon"></ion-icon>
       </div>
       <div>
+        <a href="/settings">
```
Comment (yammesicka) [inline app/templates/partials/calendar/navigation.html]: url_for

File: app/templates/settings.html [hunk 27]
```diff
@@ -0,0 +1,128 @@
...
+              <option value="1">English (US)</option>
+              <option value="2">Hebrew</option>
+            </select>
+            <p>Country</p>
+            <select class="form-select form-select-sm" id="select-form-settings" aria-label=".form-select-sm example">
+              <option value="1">Israel</option>
+              <option value="2">United States</option>
+            </select>
+            <p>Date format</p>
+            <select class="form-select form-select-sm" id="select-form-settings" aria-label=".form-select-sm example">
```
Comment (yammesicka) [inline app/templates/settings.html]: Add label

File: app/templates/settings.html [hunk 28]
```diff
@@ -0,0 +1,128 @@
...
+            </div>
+          </section>
+        </div>
+
+        <div id="settings-main" class="settings-main">
+          <!-- Language and region -->
+          <article id="tab0-content" class="tab-show">
+            <h2>Language and region</h2>
+            <p>Language</p>
+            <select class="form-select form-select-sm" id="select-form-settings" aria-label=".form-select-sm example">
```
Comment (yammesicka) [inline app/templates/settings.html]: Add label

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/static/grid_style.css [hunk 29]
```diff
@@ -94,7 +94,6 @@ header {
   position: sticky;
   top: 0;
   display: flex;
-  grid-flow: row wrap;
```
Comment (yammesicka) [inline app/static/grid_style.css:119]: You did this on purpose?

File: app/static/js/settings.js [hunk 30]
```diff
@@ -0,0 +1,23 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const tabBtn = document.querySelectorAll(".tab");
```
Comment (yammesicka) [inline app/static/js/settings.js]: document.getElementsByClassName

File: app/static/settings_style.css [hunk 31]
```diff
@@ -0,0 +1,136 @@
...
+.left-options-bar {
+  display: flex;
+}
+
+.settings-main article {
+  margin-bottom: 2em;
+}
+
+.settings-options {
+  flex: 1
```
Comment (yammesicka) [inline app/static/settings_style.css]: ;

File: app/static/settings_style.css [hunk 32]
```diff
@@ -0,0 +1,136 @@
...
+  color: #222831;
+  display: flex;
+  font-size: 0.9em;
+}
+
+#settings-left {
+  padding: 1.5em 1.5em;
+  display: flex;
+  flex-direction: column;
+  background-color: rgb(230, 230, 230)
```
Comment (yammesicka) [inline app/static/settings_style.css]: ;

**Review by yammesicka — COMMENTED** [review]
DM me after you have done fixing it

File: app/templates/settings.html [hunk 33]
```diff
@@ -0,0 +1,128 @@
...
+          </article>
+          <article id="tab3-content" class="tab-hide"></article>
+          <article id="tab4-content" class="tab-hide"></article>
+        </div>
+      </main>
+    </div>
+  </div>
+
+<!-- Script -->
+<script src="{{ url_for('static', path='/js/settings.js') }}"></script>
```
Comment (yammesicka) [inline app/templates/settings.html]: One line below

File: app/static/grid_style.css (hunk 29 above)
Comment (Eliory09) [reply app/static/grid_style.css:119]: VSC alerts it's an unknown property :)

**Review by yammesicka — APPROVED** [review]

**Pull request merged**